File: gen_b64.py
# gen_b64.py
# Convert gambar WebP → PNG → base64 untuk Figma Scripter
# Requires: pip install Pillow
#
# Usage: python gen_b64.py

import base64
import io
import os
from PIL import Image

BASE = r"c:\dev\kafetani\assets\img"
OUT  = r"c:\dev\kafetani\figma_images.js"

images = {
    # products
    "kopi_susu_gula_aren":   os.path.join(BASE, "products", "kopi_susu_gula_aren.webp"),
    "croissant_butter":      os.path.join(BASE, "products", "croissant_butter.webp"),
    "biji_kopi_arabica_gayo":os.path.join(BASE, "products", "biji_kopi_arabica_gayo.webp"),
    "kopi_lokal":            os.path.join(BASE, "products", "kopi_lokal.webp"),
    "arabica_gayo":          os.path.join(BASE, "products", "arabica_gayo.webp"),
    "gula_aren":             os.path.join(BASE, "products", "gula_aren.webp"),
    "bakeri_segar":          os.path.join(BASE, "products", "bakeri_segar.webp"),
    # about
    "about_bahan_segar":     os.path.join(BASE, "about", "bahan_segar.webp"),
    "about_petani_lokal":    os.path.join(BASE, "about", "petani_lokal.webp"),
    "about_pesan_online":    os.path.join(BASE, "about", "pesan_online.webp"),
    "about_bawa_pulang":     os.path.join(BASE, "about", "bawa_pulang.webp"),
}

lines = [
    "// AUTO-GENERATED — gambar WebP→PNG→base64 untuk Figma Scripter",
    "// Format: PNG (didukung Figma createImageAsync)",
    "const IMAGES = {",
]

for key, path in images.items():
    if not os.path.exists(path):
        print(f"SKIP (not found): {path}")
        continue

    # Buka WebP, convert ke PNG di memory
    img = Image.open(path).convert("RGBA")  # RGBA agar transparansi terjaga
    buf = io.BytesIO()
    img.save(buf, format="PNG", optimize=True)
    buf.seek(0)

    b64 = base64.b64encode(buf.read()).decode("ascii")
    data_uri = f"data:image/png;base64,{b64}"
    lines.append(f'  "{key}": "{data_uri}",')
    print(f"OK  {key}  ({len(b64)//1024} KB base64)")

lines.append("};")

with open(OUT, "w", encoding="utf-8") as f:
    f.write("\n".join(lines))

print(f"\nSelesai → {OUT}")
print(f"Total size: {os.path.getsize(OUT)//1024} KB")

File: gen_b64.ts
// gen_b64.ts — Bun script
// Convert WebP → PNG → base64 untuk Figma Scripter
// Usage: bun run gen_b64.ts
// Requires: bun add sharp

import sharp from "sharp";
import { join } from "path";

const BASE = String.raw`c:\dev\kafetani\assets\img`;
const OUT  = String.raw`c:\dev\kafetani\figma_images.js`;

const images: Record<string, string> = {
  // Products
  kopi_susu_gula_aren:    join(BASE, "products", "kopi_susu_gula_aren.webp"),
  croissant_butter:       join(BASE, "products", "croissant_butter.webp"),
  biji_kopi_arabica_gayo: join(BASE, "products", "biji_kopi_arabica_gayo.webp"),
  kopi_lokal:             join(BASE, "products", "kopi_lokal.webp"),
  arabica_gayo:           join(BASE, "products", "arabica_gayo.webp"),
  gula_aren:              join(BASE, "products", "gula_aren.webp"),
  bakeri_segar:           join(BASE, "products", "bakeri_segar.webp"),
  // About
  about_bahan_segar:      join(BASE, "about", "bahan_segar.webp"),
  about_petani_lokal:     join(BASE, "about", "petani_lokal.webp"),
  about_pesan_online:     join(BASE, "about", "pesan_online.webp"),
  about_bawa_pulang:      join(BASE, "about", "bawa_pulang.webp"),
};

const lines: string[] = [
  "// AUTO-GENERATED — WebP→PNG→base64 untuk Figma Scripter",
  "// Format: PNG (Figma hanya mendukung JPEG & PNG di createImageAsync)",
  "const IMAGES = {",
];

for (const [key, path] of Object.entries(images)) {
  const file = Bun.file(path);
  if (!(await file.exists())) {
    console.warn(`SKIP (not found): ${path}`);
    continue;
  }

  // Baca WebP lalu convert ke JPEG pakai sharp (resize ke max 800px)
  const jpegBuf = await sharp(await file.arrayBuffer())
    .resize(800, 800, { fit: "inside", withoutEnlargement: true })
    .jpeg({ quality: 80 })
    .toBuffer();

  const b64 = jpegBuf.toString("base64");
  lines.push(`  "${key}": "data:image/jpeg;base64,${b64}",`);
  console.log(`OK  ${key}  (${Math.round(b64.length / 1024)} KB)`);
}

lines.push("};");

await Bun.write(OUT, lines.join("\n"));
console.log(`\nSelesai → ${OUT}`);
console.log(`Total: ${Math.round((await Bun.file(OUT).size) / 1024)} KB`);
